language/expressions.md

```diff
@@ -720,9 +726,26 @@ result. Chained assignments are permitted.
 
 Examples:
 
+~~~puppet
+
     $a = 10
     $x = $y = 0
 
+    # from array
+    [$a, $b] = [1, 2]  # assigns 1 to $a, and 2 to $b
+
+    # from hash
+    [$a, $b] = { a => 10, b => 20, c => 30 }  # assigns 10 to $a, and 20 to $b
+
+    # from class
+    class mymodule::someclass::example($x = 100) {
+      $a = 10
+    }
+    include example
+    [$a, $x] = Class['mymodule::someclass::example']  # assigns 10 to $a, and 100 to $x
```

Comment:
hmm, so

```
define x($y) { }
x{"meh": y => 1}
notice(X["meh"]["y"])
```

works, seems you should be able to get at the parameters at least?

As you say its evaluation order dependant, but I thought so was the class case since the internal vars wouldnt be available if not included already?